src/sentry/api/bases/organization_events.py

```diff
@@ -233,6 +233,14 @@ def get_split_decision(self, has_errors, has_other_data):
         # Covers cases of !A && !B and A && B
         return DashboardWidgetTypes.ERROR_EVENTS
 
+    def save_split_decision(self, widget, has_errors, has_other_data):
+        """This can be removed once the discover dataset has been fully split"""
+        new_discover_widget_split = self.get_split_decision(has_errors, has_other_data)
+
+        if widget.discover_widget_split != new_discover_widget_split:
+            widget.discover_widget_split = new_discover_widget_split
+            widget.save()
```

Comment:
Okay. This is the logic that was on events-stats 👍🏻 Now it's shared in the base class.
<img width="547" alt="image" src="https://github.com/getsentry/sentry/assets/44410/601f417b-eae3-4442-ad29-3ec7de070ab3">